includes/class-wc-payments-account.php

```diff
@@ -643,4 +645,27 @@ private function settings_changed( $changes = [] ) {
 		$diff = array_diff_assoc( $changes, $account );
 		return ! empty( $diff );
 	}
+
+	/**
+	 * Retrieves the latest ToS agreement for the account.
+	 *
+	 * @return array|null Eiter the agreement or null if unavailable.
+	 */
+	public function get_latest_tos_agreement() {
+		$account = $this->get_cached_account_data();
+
+		if ( empty( $account ) ) {
+			return null;
+		}
+
+		if ( ! isset( $account['latest_tos_agreement'] ) ) {
+			$account = $this->refresh_account_data();
+		}
+
+		if ( empty( $account ) ) {
+			return null;
+		}
```

Comment:
Account is cached for 2 hours max, I think it's fine to wait until cache is expired.

Reply:
The point here is that the `latest_tos_agreement` property should be present in the server response. If it's not, it means the client got updated and a cache refresh is necessary. Do you think this additional check is an issue in any way?

Reply:
As you suggested in slack conversation we should not rely too much on `latest_tos_agreement` always available on the server response. So I'd expect the method looks like this:
```php
	public function get_latest_tos_agreement() {
		$account = $this->get_cached_account_data();
		return ! empty( $account ) && isset( $account['latest_tos_agreement'] ) ? $account['latest_tos_agreement'] : null;
	}
```